src/Cp/Settings.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CraftCms\Cms\Cp;
+
+use craft\events\RegisterCpSettingsEvent;
+use craft\helpers\Cp as CpHelper;
+use CraftCms\Cms\Cms;
+use CraftCms\Cms\Plugin\Plugins;
+use Illuminate\Support\Facades\Event;
+
+use function CraftCms\Cms\t;
+
+class Settings
+{
+    /**
+     * Returns the list of settings.
+     */
+    public static function all(): array
+    {
+        $readOnly = ! Cms::config()->allowAdminChanges;
+        $settings = [];
+
+        $label = t('System');
+
+        $settings[$label]['general'] = [
+            'url' => route('craft.cp.settings.general.index'),
+            'icon' => 'light/sliders',
+            'label' => t('General'),
+        ];
+
+        $settings[$label]['sites'] = [
+            'icon' => sprintf('light/%s', CpHelper::earthIcon()),
+            'label' => t('Sites'),
+        ];
+
+        if (! Cms::config()->headlessMode) {
+            $settings[$label]['routes'] = [
+                'icon' => 'light/signs-post',
+                'label' => t('Routes'),
+            ];
+        }
+
+        $settings[$label]['users'] = [
+            'icon' => 'light/user-group',
+            // 'iconMask' => '@craftcms/resources/icons/light/user-group.svg',
+            'label' => t('Users'),
+        ];
+        if (Cms::config()->allowAdminChanges) {
+            $settings[$label]['addresses'] = [
+                'icon' => 'light/map-location',
+                'label' => t('Addresses'),
+            ];
+        }
+        $settings[$label]['email'] = [
+            'icon' => 'light/envelope',
+            'label' => t('Email'),
+        ];
+        $settings[$label]['plugins'] = [
+            'icon' => 'light/plug',
+            'label' => t('Plugins'),
+        ];
+
+        $label = t('Content');
+
+        $settings[$label]['sections'] = [
+            'icon' => 'light/newspaper',
+            'label' => t('Sections'),
+        ];
+        $settings[$label]['entry-types'] = [
+            'icon' => 'light/files',
+            'label' => t('Entry Types'),
+        ];
+        $settings[$label]['fields'] = [
+            'icon' => 'light/pen-to-square',
+            'label' => t('Fields'),
+        ];
+
+        $label = t('Media');
+
+        $settings[$label]['assets'] = [
+            'icon' => 'light/image',
+            'label' => t('Assets'),
+        ];
+        $settings[$label]['filesystems'] = [
+            'icon' => 'light/folder-open',
+            'label' => t('Filesystems'),
+        ];
+
+        $label = t('Plugins');
+
+        $pluginsService = app(Plugins::class);
```

Comment:
Can move this to the constructor and inject the `Settings` class in the controller method where needed. That way everything gets dependency injected. The `GeneralConfig` can also be injected into the constructor here then

That way the `all()` method doesn't need to be static either.